docs: Add example for loading ESM from CommonJS (#1236)
node-fetch/node-fetch: docs/v3-UPGRADE-GUIDE.md

# Breaking Changes

## Minimum supported Node.js version is now 12.20

> Since Node.js will deprecate version 8 at the end of 2019, we decided that node-fetch v3.x will not only drop support for Node.js 4 and 6 (which were supported in v2.x), but also for Node.js 8. We strongly encourage you to upgrade, if you still haven't done so. Check out Node.js' official [LTS plan] for more information on Node.js' support lifetime.

Reviewer: ```suggestion
Since Node.js deprecated version 10 in May 2020, we decided that node-fetch v3.x will drop support for Node.js 4, 6, 8, and 10 (which were supported in v2.x). We strongly encourage you to upgrade, if you still haven't done so. Check out Node.js' official [LTS plan] for more information on Node.js' support lifetime.
```

> Since Node.js deprecated version 10 in May 2020, we decided that node-fetch v3.x will drop support for Node.js 4, 6, 8, and 10 (which were supported in v2.x). We strongly encourage you to upgrade, if you still haven't done so. Check out Node.js' official [LTS plan] for more information on Node.js' support lifetime.

Reviewer 2: ```suggestion
Since Node.js 10 has been deprecated since May 2020, we have decided that node-fetch v3 will drop support for Node.js 4, 6, 8, and 10 (which were previously supported). We strongly encourage you to upgrade if you still haven't done so. Check out the Node.js official [LTS plan] for more information.
```

## Converted to ES Module

> This module was converted to be a ESM only package in version `@3.0.0-beta.10`

Reviewer 2: ```suggestion
This module was converted to be a ESM only package in version `3.0.0-beta.10`.
```

Using require to load an ES module is not supported because ES modules have asynchronous execution. Instead, use import() to load an ES module from a CommonJS module.

```js

> // mod.cjs
> const fetch = (...args) => import('node-fetch').then(mod => mod.default(...args));

Reviewer 2: Synchronise

```suggestion
const fetch = (...args) => import('node-fetch').then(({default: fetch}) => fetch(...args));
```

```

## The `timeout` option was removed.

> Since this was never part of the fetch specification, it was removed. AbortSignal offers a more fine grained control of request timeouts, and is standardized in the Fetch spec. For convenience, you can use [timeout-signal](https://github.com/Richienb/timeout-signal) as a workaround:

Reviewer 2: ```suggestion
Since this was never part of the fetch specification, it was removed. AbortSignal offers more fine grained control of request timeouts, and is standardized in the Fetch spec. For convenience, you can use [timeout-signal](https://github.com/Richienb/timeout-signal) as a workaround:
```
